storage/src/test/java/org/apache/kafka/tiered/storage/integration/PartitionsExpandTest.java

```diff
@@ -0,0 +1,106 @@
+        final Integer broker1 = 1;
+        final String topicA = "topicA";
+        final Integer p0 = 0;
+        final Integer p1 = 1;
+        final Integer p2 = 2;
+        final Integer partitionCount = 1;
+        final Integer replicationFactor = 2;
+        final Integer maxBatchCountPerSegment = 1;
+        final boolean enableRemoteLogStorage = true;
+        final List<Integer> p0Assignment = Arrays.asList(broker0, broker1);
+        final List<Integer> p1Assignment = Arrays.asList(broker0, broker1);
+        final List<Integer> p2Assignment = Arrays.asList(broker1, broker0);
+
+        builder
+                .createTopic(topicA, partitionCount, replicationFactor, maxBatchCountPerSegment,
+                        Collections.singletonMap(p0, p0Assignment), enableRemoteLogStorage)
+                // produce events to partition 0
+                .expectSegmentToBeOffloaded(broker0, topicA, p0, 0, new KeyValueSpec("k0", "v0"))
+                .expectSegmentToBeOffloaded(broker0, topicA, p0, 1, new KeyValueSpec("k1", "v1"))
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 2L)
+                .produce(topicA, p0, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+                // expand the topicA partition-count to 3
+                .createPartitions(topicA, 3, mkMap(mkEntry(p1, p1Assignment), mkEntry(p2, p2Assignment)))
+                // consume from the beginning of the topic to read data from local and remote storage for partition 0
+                .expectFetchFromTieredStorage(broker0, topicA, p0, 2)
+                .consume(topicA, p0, 0L, 3, 2)
+
+                .expectLeader(topicA, p1, broker0, false)
+                .expectLeader(topicA, p2, broker1, false)
+
+                // produce events to partition 1
+                .expectSegmentToBeOffloaded(broker0, topicA, p1, 0, new KeyValueSpec("k0", "v0"))
+                .expectSegmentToBeOffloaded(broker0, topicA, p1, 1, new KeyValueSpec("k1", "v1"))
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p1, 2L)
+                .produce(topicA, p1, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+
+                // produce events to partition 2
+                .expectSegmentToBeOffloaded(broker1, topicA, p2, 0, new KeyValueSpec("k0", "v0"))
+                .expectSegmentToBeOffloaded(broker1, topicA, p2, 1, new KeyValueSpec("k1", "v1"))
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p2, 2L)
+                .produce(topicA, p2, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+
+                // produce some more events to partition 0 and expect the segments to be offloaded
+                // NOTE: Support needs to be added to capture the offloaded segment event for already sent message (k2, v2)
```

Comment:
we risk forgetting this commented line. please add a JIRA

Reply:
Filed KAFKA-15431 to track this.